pkg/cli/cli_e2e_test.go

```diff
@@ -0,0 +1,165 @@
+package cli
+
+import (
+	"flag"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/openshift/must-gather-clean/pkg/obfuscator"
+	"github.com/openshift/must-gather-clean/pkg/reporting"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	configOpenshiftDefaultPath = "examples/openshift_default.yaml"
+)
+
+func TestEndToEnd(t *testing.T) {
+	var input, report string
+	fs := flag.NewFlagSet("e2e-fs", flag.ContinueOnError)
+	fs.StringVar(&input, "input", "", "")
+	fs.StringVar(&report, "report", "", "")
+	if err := fs.Parse(flag.Args()); err != nil {
+		t.Fatal(err)
+	}
+	if input == "" || report == "" {
+		t.Fatal("Expected arguments --input && --report")
+	}
+
+	// find the path to the project root of this file
+	_, filename, _, _ := runtime.Caller(0)
+	rootDir := path.Join(path.Dir(filename), "..", "..")
+	// relative paths -> absolute paths
+	inputDir := path.Join(rootDir, input)
+	outputDir := path.Join(rootDir, fmt.Sprintf("%s.cleaned", input))
+	configPath := path.Join(rootDir, configOpenshiftDefaultPath)
+	generatedReportDir := path.Join(rootDir, fmt.Sprintf("%s-report", input))
+	reportPath := path.Join(rootDir, report)
+
+	err := Run(configPath,
+		inputDir,
+		outputDir,
+		true,
+		generatedReportDir,
+		runtime.NumCPU())
+	require.NoError(t, err)
+
+	// read reports
+	truthReport := readReport(t, reportPath)
+	generatedReport := readReport(t, filepath.Join(generatedReportDir, reportFileName))
+	removeRelativePath(generatedReport, inputDir)
+	// compare reports
+	verifyReport(t, inputDir, truthReport, generatedReport)
+	verifyObfuscation(t, outputDir, generatedReport)
+}
+
+func removeRelativePath(r *reporting.Report, path string) {
+	for i := range r.Omissions {
+		r.Omissions[i] = strings.TrimPrefix(r.Omissions[i], path)
+	}
+}
+
+func verifyObfuscation(t *testing.T, dir string, report *reporting.Report) {
+	// report should already be verified by verifyReport before to ensure it does contain correct information
+	generatedMap := map[string]string{}
```

Comment:
you made that generic now, but the assertions mention "IP" everywhere, I think that needs some updates